src/network/NetworkController.ts

```diff
@@ -78,36 +68,125 @@ export interface NetworkConfig extends BaseConfig {
  * @property isCustomNetwork - Identifies if the network is a custom network
  * @property provider - RPC URL and network name provider settings
  */
-export interface NetworkState extends BaseState {
+export type NetworkState = {
   network: string;
-  isCustomNetwork: boolean;
+  isCustomNetwork?: boolean;
```

Comment:
Why are these properties optional now? It seems like it would save us from having to check whether they exist before we use them, if they were not optional.

Reply:
They are optional because we have a default state which includes values for each.